### PR TITLE
More Python 3 support and beginning of unencrypted session handling 

### PR DESCRIPTION
**Pull request by tvincentNuoDB**
Jira issue [DB-10681](http://tools/jira/browse/DB-10681): Python 3 handling added, there was a lot of issues since the previous version abused the ascii string or bytes type in python 2. As these are two separate datatypes (and strings are Unicode), a lot of changes had to be made to crypt.py. Some minor changes to test cases for Python 3 as well. Connections to the server can be established and data can be sent, however receiving SQL query data currently causes issues 

Jira issue [DB-5725](http://tools/jira/browse/DB-5725): The server can now send messages over the wire in an unencrypted manor with options 'cipher':'None'. However in accordance to NuoDB standards, the handshake to authenticate is still encrypted with RC4. 

Some general cleanup and logical rewrites were also applied.

### CODE REVIEW

File: pynuodb/crypt.py [hunk 1]
```diff
@@ -235,27 +261,44 @@ def computeSessionKey(self, clientKey, verifier):
...
             state[i], state[j] = state[j], state[i]
 
     def transform(self, data):
         transformed = []
-        state = self.__S
-
+        state = self.__state
+        if type(data) is bytes:
+            data.decode("latin-1")
+        print("Data is ")
```
Comment (mjmichaels) [inline pynuodb/crypt.py]: Should these print statements be removed?

**Comment by tvincentNuoDB** [comment]
Whoops, removed the print statements.

File: pynuodb/session.py [hunk 2]
```diff
@@ -67,7 +66,7 @@ def __str__(self):
 class Session:
 
     __AUTH_REQ = "<Authorize TargetService=\"%s\" Type=\"SRP\"/>"
-    __SRP_REQ = "<SRPRequest ClientKey=\"%s\" Cipher=\"RC4\" Username=\"%s\"/>"
+    __SRP_REQ = "<SRPRequest ClientKey=\"%s\" Cipher=\"RC4\" Username=\"%s\"/>" #Why is this hard coded...
```
Comment (mjmichaels) [inline pynuodb/session.py]: Do you mean to add this comment? Is your question why is it hardcoded to RC4?

Comment (tvincentNuoDB) [inline pynuodb/session.py]: It was meant to be a note to myself but, didn't get around to resolving before pushing. I think this should be dynamically built from the options passed into the connection

**Pull request merged**